components/cldr-json-data-provider/src/reader.rs

```diff
@@ -0,0 +1,12 @@
+use crate::error::Error;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+pub fn open_reader(path: PathBuf) -> Result<BufReader<File>, Error> {
```

Comment:
Is this set of PRs (198, 199, 200) attempting to address the concern that data provider implementations should also decouple the "where" and the "what" of the data?  Or is it not currently in scope here?

When we do get to that, I imagine that we'll want to stick to the Rust equivalent of the Stream interface (whatever that is) without using specific types, ex: `BufReader` instead of `BufReader<File>`.